File: week-02/day-01/draw_diamond.py
# Write a program that reads a number from the standard input, then draws a
# diamond like this:
#
#
#    *
#   ***
#  *****
# *******
#  *****
#   ***
#    *
#
# The diamond should have as many lines as the number was
a = int(input("Please enter a number: "))

for i in range(1,(a+1)//2+1):
    print(" "*((a+1)//2-i)+"*"*(i*2-1)+" "*((a+1)//2-i))
if a%2 != 0:   
    for i in range(1,a//2+1):
        print(" "*i+"*"*(a-i*2)+" "*i)
else:   
    for i in range(1,a//2+1):
        print(" "*(i-1)+"*"*((a+1)-i*2)+" "*(i-1))

File: week-07/day-02/draw_diamond.js
'use strict';

let lineCount = 17;

// Write a program that draws a
// diamond like this:
//
//
//    *
//   ***
//  *****
// *******
//  *****
//   ***
//    *
//
// The diamond should have as many lines as lineCount is

for (let i = 1; i < Math.floor((lineCount+1)/2+1); ++i) {
    console.log(" ".repeat(Math.floor((lineCount+1)/2-i)) + "*".repeat(i*2-1));
}
for (let i = 1; i < Math.floor(lineCount/2+1); ++i) {
    if (lineCount%2 !== 0) {
        console.log(" ".repeat(i) + "*".repeat(lineCount-i*2));
    } else {
        console.log(" ".repeat(i-1) + "*".repeat((lineCount+1)-i*2));
    }
} 
